### PR TITLE
removed_unneeded_api_calls

### PR DESCRIPTION
**Pull request by abedelaziz-deriv**
- Removed hidden api calls.
- Converted the project into a package.
- Fixed Unit tests.

### CODE REVIEW

File: api_builder.dart [hunk 1]
```diff
@@ -49,6 +49,47 @@ class APIBuilder extends Builder {
...
+    'affiliate_register_person',
+    'available_accounts',
+    'cashier_payments',
+    'cashier_withdrawal_cancel',
+    'change_email',
+    'change_password',
+    'get_account_types',
+    'jtoken_create',
+    'link_wallet',
+    'new_account_wallet',
```
Comment (ramin-deriv) [inline api_builder.dart:68]: These ones DerivGo uses:
new_accont_wallet
reset_password

File: api_builder.dart [hunk 2]
```diff
@@ -49,6 +49,47 @@ class APIBuilder extends Builder {
...
+    'request_report',
+    'reset_password',
+    'service_token',
+    'trading_platform_accounts',
+    'trading_platform_asset_listing',
+    'trading_platform_available_accounts',
+    'trading_platform_deposit',
+    'trading_platform_investor_password_change',
+    'trading_platform_leverage',
+    'trading_platform_new_account',
```
Comment (ramin-deriv) [inline api_builder.dart:86]: get_service_token
and trading platform calls are used for DerivEZ in DerivGO.
DerivEZ is going to be removed from DerivGO. I think we can hide these later when we removed their code from DerivGO first.

File: lib/api/response/cashier_payments_response_extended.dart
Comment (ramin-deriv) [inline lib/api/response/cashier_payments_response_extended.dart:1]: these classes that end with _extended, are removed, they are used in DerivGO

File: api_builder.dart (hunk 1 above)
Comment (ramin-deriv) [reply api_builder.dart:68]: service_token
available_accounts
wallet_migration

File: api_builder.dart [hunk 3]
```diff
@@ -49,6 +49,42 @@ class APIBuilder extends Builder {
...
 
+  static const List<String> hiddenCalls = <String>[
+    'account_closure',
+    'account_security',
+    'account_statistics',
+    'affiliate_account_add',
+    'affiliate_add_company',
+    'affiliate_add_person',
+    'affiliate_register_person',
+    'cashier_withdrawal_cancel',
```
Comment (ramin-deriv) [inline api_builder.dart:60]: I think we also will need this in DerivGO in the near future. in native cashier withdrawal.

File: lib/api/response/trading_platform_password_reset_response_extended.dart
Comment (ramin-deriv) [inline lib/api/response/trading_platform_password_reset_response_extended.dart:1]: This file also needs to be retained.

File: lib/api/response/available_accounts_response_extended.dart
Comment (ramin-deriv) [inline lib/api/response/available_accounts_response_extended.dart:1]: this as well

**Comment by abedelaziz-deriv** [comment]
This PR is not needed according to this discussion https://deriv-group.slack.com/archives/C01T300JQN4/p1687850869659729

**Pull request closed without merging**